doc/reference/reference_lua/config.rst

```diff
@@ -205,6 +227,65 @@ config API
                   alerts: []
                 ...
 
+    .. _config_api_reference_instance_uri:
+
+    ..  method:: instance_uri([uri_type, opts])
+
+        **Since:** :doc:`3.1.0 </release/3.1.0>`
+
+        Get a URI of the current or remote instance.
+
+        :param string uri_type: a URI type. There are the following URI types are supported:
+
+                                * ``peer`` -- a URI used to advertise the instance to other cluster members. See also: :ref:`iproto.advertise.peer <configuration_reference_iproto_advertise_peer>`.
+                                * ``sharding`` -- a URI used to advertise the current instance to a router and rebalancer. See also: :ref:`iproto.advertise.sharding <configuration_reference_iproto_advertise_sharding>`.
+
+        :param table opts: options to pass. The following options are available:
+
+                           -   ``instance`` -- the name of a remote instance whose URI should be obtained
+
+        :return: an instance URI
+
+        **Example**
+
+        The example below shows how to get a URI used to advertise ``storage-b-003`` to other cluster members:
+
+        ..  code-block:: lua
+
+            local config = require('config')
+            config:instance_uri('peer', { instance = 'storage-b-003' })
+
+
+    .. _config_api_reference_instances:
+
+    ..  method:: instances()
+
+        **Since:** :doc:`3.1.0 </release/3.1.0>`
+
+        List all instances of the cluster.
+
+        :return: a table containing information about instances
+
+        The returned table uses instance names as the keys and contains the following information for each instance:
+
+        -   ``instance_name`` -- an instance name
+        -   ``replicaset_name`` -- the name of a replica set the instance belongs to
+        -   ``group_name`` -- the name of a group the instance belongs to
+
+        **Example**
+
+        The example below shows how to use ``instances()`` to get the names of all instances in the cluster, create a connection to each instance using the :ref:`connpool <connpool_module>` module, and log connection URIs using the :ref:`log <log-module>` module:
+
+        ..  code-block:: lua
+
+            local config = require('config')
+            for instance_name in pairs(config:instances()) do
+                local connpool = require('experimental.connpool')
+                local conn = connpool.connect(instance_name)
+                local log = require('log')
+                log.info(string.format("Connection URI for '%s': %s:%s", instance_name, conn.host, conn.port))
```

Comment:
BTW, `%q` can be used instead of `"%s"` for quoting.

```lua
tarantool> log.info('xxx %q zzz', 'yyy')
xxx "yyy" zzz
```

And an explicit `string.format` is not needed, `log.info` calls it itself.